test/integration/scheduler/queue_test.go

```diff
@@ -805,28 +805,34 @@ func TestCoreResourceEnqueue(t *testing.T) {
 		{
 			name: "Pods with PodTopologySpread should be requeued when a Node is updated to have the topology label",
 			initialNodes: []*v1.Node{
-				st.MakeNode().Name("fake-node1").Label("node", "fake-node").Obj(),
-				st.MakeNode().Name("fake-node2").Label("node", "fake-node").Obj(),
+				st.MakeNode().Name("fake-node1").Label("node", "fake-node").Label("region", "fake-node").Obj(),
+				st.MakeNode().Name("fake-node2").Label("node", "fake-node").Label("region", "fake-node").Obj(),
 			},
 			initialPods: []*v1.Pod{
 				st.MakePod().Name("pod1").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node1").Obj(),
 				st.MakePod().Name("pod2").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node2").Obj(),
+				st.MakePod().Name("pod3").Label("key1", "val").SpreadConstraint(1, "region", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node2").Obj(),
 			},
 			pods: []*v1.Pod{
-				// - Pod3 and Pod4 will be rejected by the PodTopologySpread plugin.
-				st.MakePod().Name("pod3").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
-				st.MakePod().Name("pod4").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+				// - Pod4, Pod5, Pod6, Pod7 will be rejected by the PodTopologySpread plugin.
```

Comment:
- pod4: match -> unmatch      requeue
- pod5: unmatch -> match      requeue
- pod6: match -> match, but labels changes    requeue
- pod7: unmatch -> unmatch   not requeue

match -> match but not requeue 
It is difficult to implement in this case because the label will always change.... @sanposhiho

Reply:
> pod6: match -> match, but labels changes (requeue)

The region label value isn't changed actually. Should we?

> It is difficult to implement in this case because the label will always change....

You have to add a new label (e.g., `node-pool`) that isn't changed so that each label will be like:

- node label: will be removed after the update
- zone label: will be added after the update
- region label: the value will be changed after the update (as said above, currently not though, I guess it's a mistake)
- node-label label: no change at all after the update.

Reply:
> The region label value isn't changed actually. Should we?
> 
> >

region label isn't changed, but whole labels are different
we have this logic: 
```go
if modifiedNodeMatching && !equality.Semantic.DeepEqual(originalNode.Labels, modifiedNode.Labels) {
			logger.V(5).Info("the node is updated with different label values but still matches all pod topology spread constraints, and the pod may be schedulable now",
				"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
			return framework.Queue, nil
		}
```

Reply:
Oh oh, I see. Then, shouldn't we change it to only check the update(s) of topologyKey label(s)?
So, when Pod has `hoge` label in topologyKey, then we only check if the value of `hoge` label is updated or not, we never pay attention to other labels.

Reply:
> > The region label value isn't changed actually. Should we?
> > >
> 
> region label isn't changed, but whole labels are different we have this logic:
> 
> ```go
> if modifiedNodeMatching && !equality.Semantic.DeepEqual(originalNode.Labels, modifiedNode.Labels) {
> 			logger.V(5).Info("the node is updated with different label values but still matches all pod topology spread constraints, and the pod may be schedulable now",
> 				"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
> 			return framework.Queue, nil
> 		}
> ```

So I need to remove this logic, right?

Reply:
So, we have to remove the comparison of the **whole** labels, and instead have to check if there's any value update(s) in the label specified as topologyKey of any spread constraints.

Reply:
done